homeassistant/components/reolink/siren.py

```diff
@@ -0,0 +1,111 @@
+"""This component provides support for Reolink siren entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import Host
+
+from homeassistant.components.siren import (
+    ATTR_DURATION,
+    ATTR_VOLUME_LEVEL,
+    SirenEntity,
+    SirenEntityDescription,
+    SirenEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSirenEntityDescriptionMixin:
+    """Mixin values for Reolink siren entities."""
+
+    method: Callable[[Host, int, bool, int], Any]
+    volume: Callable[[Host, int, int], Any]
+
+
+@dataclass
+class ReolinkSirenEntityDescription(
+    SirenEntityDescription, ReolinkSirenEntityDescriptionMixin
+):
+    """A class that describes siren entities."""
+
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+
+
+SIREN_ENTITIES = (
+    ReolinkSirenEntityDescription(
+        key="siren",
+        name="Siren",
+        icon="mdi:alarm-light",
+        supported=lambda api, ch: api.supported(ch, "siren"),
+        method=lambda api, ch, on_off, duration: api.set_siren(
+            ch, on_off, int(duration)
+        ),
```

Comment:
lambdas shouldn't be multiline I've been told before. You can provide a method instead

Reply:
I changed it to a seprate method, thanks for the feedback.
Could you review again/approve?